src/components/Banners/SlackNotificationBanner.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useState } from "react";
+import { useMutation, useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Icon from "@leafygreen-ui/icon";
+import IconButton from "@leafygreen-ui/icon-button";
+import { uiColors } from "@leafygreen-ui/palette";
+import { Body } from "@leafygreen-ui/typography";
+import { Popconfirm, Input } from "antd";
+import Cookies from "js-cookie";
+import { Banners, styles } from "components/Banners";
+import { PageWrapper } from "components/styles";
+import {
+  useBannerDispatchContext,
+  useBannerStateContext,
+} from "context/banners";
+import {
+  UpdateUserSettingsMutation,
+  UpdateUserSettingsMutationVariables,
+  GetUserSettingsQuery,
+} from "gql/generated/types";
+import { UPDATE_USER_SETTINGS } from "gql/mutations";
+import { GET_USER_SETTINGS } from "gql/queries";
+import { withBannersContext } from "hoc/withBannersContext";
+
+const { Banner } = styles;
+
+const slackNotificationBannerCookieKey = "slack-notification-banner";
+const doNotShowslackNotificationBannerCookie = "false";
+
+export const SlackNotificationBannerCore = () => {
+  const dispatchBanner = useBannerDispatchContext();
+  const bannersState = useBannerStateContext();
+
+  const [slackUsername, setSlackUsername] = useState("");
```

Comment:
I already have a slackUsername defined in my userSettings this should be prefilled if it exists.